Remotion/Security/UnitTests/XmlAsserter/XmlElementConstraint.cs

```diff
@@ -25,26 +25,27 @@ public class XmlElementConstraint : Constraint  {
     private readonly string _expectedNamespace;
     private readonly string _expectedLocalName;
 
+    public override string Description => "Test";
+
     public XmlElementConstraint (string expectedNamespace, string expectedLocalName)
     {
       _expectedNamespace = expectedNamespace;
       _expectedLocalName = expectedLocalName;
     }
 
-    public override bool Matches (object actual)
+    public override ConstraintResult ApplyTo<TActual> (TActual actual)
+    {
+      return new XmlElementConstraintResult(this, actual, Matches(actual));
+    }
+
+    private bool Matches (object actual)
     {
-      base.actual = actual;

```

Comment:
Don't know what this did before, but do we not need to have something equivalent any more? Am fine either way, just want to understand the reasoning.

Reply:
the constraint concept changed so a constraint is more like an immutable function that is constructed once and can be used multiple times. All information regarding a specific use is returned using the ConstraintResult, where actual is set using the ctor.